auth/token/token.go

```diff
@@ -36,14 +38,91 @@ type Token = jwt.Token
 // Claims is a JWT Claims object
 type Claims struct {
 	*jwt.StandardClaims
-	Username string `json:"username"`
+	ProfileID string `json:"profileID"`
 }
 
 // Parse will parse, validate and return a token
 func Parse(tokenString string, tokens Source) (*Token, error) {
 	return jwt.Parse(tokenString, tokens.VerificationKey)
 }
 
+// NewPrivKeyAuthToken creates a JWT token string suitable for making requests
+// authenticated as the given private key
+func NewPrivKeyAuthToken(pk crypto.PrivKey, ttl time.Duration) (string, error) {
+	signingMethod, err := jwtSigningMethod(pk)
+	if err != nil {
+		return "", err
+	}
+
+	t := jwt.New(signingMethod)
+
+	id, err := key.IDFromPrivKey(pk)
+	if err != nil {
+		return "", err
+	}
+
+	rawPrivBytes, err := pk.Raw()
+	if err != nil {
+		return "", err
+	}
+	signKey, err := x509.ParsePKCS1PrivateKey(rawPrivBytes)
+	if err != nil {
+		return "", err
+	}
+
+	var exp int64
+	if ttl != time.Duration(0) {
+		exp = Timestamp().Add(ttl).In(time.UTC).Unix()
+	}
+
+	// set our claims
+	t.Claims = &Claims{
+		StandardClaims: &jwt.StandardClaims{
+			Issuer:  id,
```

Comment:
These are not completely compatible with the cloud side so far, but I'm happy if we have this work for node to node comms before aligning those two.